### ISSUE TITLE
Is there a way to see if applicationinsights has been initialised

### ISSUE THREAD

**Issue by grug** [microsoft/ApplicationInsights-node.js#1064]
Hi there,

Is there a way to see if applicationinsights has been initalised? For instance, I have the following code:

```ts
    import * as appInsights from 'applicationinsights';

    appInsights.setup(applicationInsightsConnectionString);

    appInsights.start();
```

Later on in my code, I want to check to see if appInsights.start() has been called so I know that I should either call functions like `trackException` or `trackTrace` or warn the user that they haven't initialised things.

I had a look through the docs and it doesn't seem like there's a flag or way to determine this. Is my understanding correct and I'll have to track this myself?

**Comment by hectorhdzg** [comment]
We use internal properties to track this so initialization works correctly and to prevent multiple starts, but we don't expose them, can you elaborate on your scenario?, setup should prepare everything for manual tracking, like track APIs that you mention, start will do auto instrumentation stuff, maybe you can check for appInsights.defaultClient to exist and should be enough.

**Comment by grug** [comment]
You're right, though, checking for `defaultClient` existing would do the job :)
